diff --git a/core/modules/user/src/Tests/UserRegistrationTest.php b/core/modules/user/src/Tests/UserRegistrationTest.php
index 1e16bc7..84e956e 100644
--- a/core/modules/user/src/Tests/UserRegistrationTest.php
+++ b/core/modules/user/src/Tests/UserRegistrationTest.php
@@ -209,7 +209,7 @@ public function testUuidFormState() {
 
     $user_storage = \Drupal::entityManager()->getStorage('user');
 
-    $this->assertTrue($user_storage->loadByProperties(['name' => $edit['name']]));
+    $this->assertTrue($user_storage->loadByProperties(['name' => $edit['name']]), SafeMarkup::format('User with name @name loaded.', ['@name' => $edit['name']]));
     $this->drupalLogout();
 
     // Create a second account.
@@ -220,7 +220,7 @@ public function testUuidFormState() {
     $this->drupalPostForm('user/register', $edit, t('Create new account'));
     $this->assertResponse(200);
 
-    $this->assertTrue($user_storage->loadByProperties(['name' => $edit['name']]));
+    $this->assertTrue($user_storage->loadByProperties(['name' => $edit['name']]), SafeMarkup::format('User with name @name loaded.', ['@name' => $edit['name']]));
   }
 
   function testRegistrationDefaultValues() {
diff --git a/core/modules/user/user.module b/core/modules/user/user.module
index 3d76074..e99f883 100644
--- a/core/modules/user/user.module
+++ b/core/modules/user/user.module
@@ -1424,3 +1424,9 @@ function template_preprocess_user(&$variables) {
     $variables['content'][$key] = $variables['elements'][$key];
   }
 }
+
+function user_modules_installed($modules) {
+  if (in_array('user', $modules)) {
+    \Drupal::getContainer()->get('module_installer')->install(array('contacts'), TRUE);
+  }
+}
